sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -376,6 +400,170 @@ private static MessagingEntityType validateEntityPaths(ClientLogger logger, Stri
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #topicName A topic name should be set as well.
+         */
+        public ServiceBusMultiSessionProcessorClientBuilder subscriptionName(String subscriptionName) {
+            this.subscriptionName = subscriptionName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic.
+         *
+         * @param queueName Name of the queue.
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #subscriptionName A subscription name should be set as well.
+         */
+        public ServiceBusMultiSessionProcessorClientBuilder queueName(String queueName) {
+            this.queueName = queueName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic.
+         *
+         * @param topicName Name of the topic.
+         *
+         * @return The modified {@link ServiceBusMultiSessionProcessorClientBuilder} object.
+         * @see #subscriptionName A subscription name should be set as well.
+         */
+        public ServiceBusMultiSessionProcessorClientBuilder topicName(String topicName) {
+            this.topicName = topicName;
+            return this;
+        }
+
+        public ServiceBusMultiSessionProcessorClientBuilder maxConcurrentSessions(int maxConcurrentSessions) {
+            this.maxConcurrentSessions = maxConcurrentSessions;
+            return this;
+        }
+
+        /**
+         * The function that is called for each message received by this {@link ServiceBusMultiSessionProcessorClient}.
+         * The input contains the {@link ServiceBusReceivedMessage} and {@lnk SessionManager}.
+         *
+         * @param processMessage The callback that's called when a message is received by this {@link ServiceBusMultiSessionProcessorClient}.
+         * @return The updated {@link ServiceBusMultiSessionProcessorClientBuilder} instance.
+         * @throws NullPointerException if {@code processEvent} is {@code null}.
+         */
+        public ServiceBusMultiSessionProcessorClientBuilder processMessage(BiConsumer<ServiceBusReceivedMessage, SessionMessageManager> processMessage) {
+            this.processMessage = Objects.requireNonNull(processMessage, "'processMessage' cannot be null");
+            return this;
+        }
+
+        /**
+         * The function that is called when an error occurs while processing messages. The input contains exception
+         * when the error happened.
+         *
+         * @param processError The callback that's called when an error occurs while processing messages.
+         * @return The updated {@link ServiceBusMultiSessionProcessorClientBuilder} instance.
+         */
+        public ServiceBusMultiSessionProcessorClientBuilder processError(Consumer<ServiceBusErrorContext> processError) {
```

Comment:
Would `ServiceBusErrorContext` contain information like the session ID or something of that nature?